lib/streamlit/runtime/connection_factory.py

```diff
@@ -47,13 +48,32 @@ def _create_connection(
     connection_class must be a concrete type. The public-facing connection API allows
     the user to specify the connection class to use as a string literal for convenience.
     """
-    return connection_class(
-        connection_name=name,
-        **kwargs,
+    return connection_class(connection_name=name, **kwargs)
+
+
+# NOTE: Adding support for a new first party connection requires:
+#   1. Adding the new connection name and class to this function.
+#   2. Writing a new @overload signature mapping the connection's name to its class.
+def _get_first_party_connection(connection_name: str):
+    FIRST_PARTY_CONNECTIONS = {"sql"}
+
+    if connection_name == "sql":
+        return SQL
+
+    raise StreamlitAPIException(
+        f"Invalid connection {connection_name}. "
+        f"Supported connection classes: {FIRST_PARTY_CONNECTIONS}"
     )
 
 
-@overload  # type: ignore
+@overload
+def connection_factory(
+    connection_class: Literal["sql"], name: str = "default", **kwargs
```

Comment:
nit: The `connect` function has an explicit parameter `autocommit: bool = False`, maybe it makes sense to add this to this overload signature as well

Reply:
Ah, yeah -- we should definitely do this